sklearn/linear_model/stochastic_gradient.py

```diff
@@ -714,9 +714,14 @@ def __init__(self, loss="hinge", penalty='l2', alpha=0.0001, l1_ratio=0.15,
             average=average)
 
     def _check_proba(self):
-        check_is_fitted(self, "t_")
-
-        if self.loss not in ("log", "modified_huber"):
+        try:
+            check_is_fitted(self, ["t_", "coef_", "intercept_"],
+                            all_or_any=all)
+            is_fitted = True
+        except _NotFittedError:
+            is_fitted = False            
```

Comment:
If I remove the call, GridSearchCV would  throw loss hinge error when predict_proba is called. The problem came from [this line](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/metaestimators.py#L35). It tried to get the predict_proba attribute of SGDClassifier class but the default losss is hinge
In the first commit I pushed, I removed the call but it failed some tests: